client/src/components/Viewer/Viewer.js

```diff
@@ -37,169 +40,188 @@ function Viewer() {
         setUploadPercentage(prevValue => ({
           ...prevValue,
           [fileName]: {
             ...prevValue[fileName],
-            minio: per,
+            dzsave: Number(per),
           },
         }));
-      }
-    });
-
-    socket.on('error', error => {
-      console.error('Socket.IO error:', error);
-    });
-
-    socket.on('dzsave-progress', progress_data => {
-      let per = progress_data.progress;
-      const fileName = currentFile.name.name;
-      setUploadPercentage(prevValue => ({
-        ...prevValue,
-        [fileName]: {
-          ...prevValue[fileName],
-          dzsave: Number(per),
-        },
-      }));
-    });
+      });
 
-    socket.on('AddedUser', async () => {
-      const formData = new FormData();
-      setDisplayProgressBar(true);
-      formData.append('file', currentFile.name);
-      let bucketURL = config.BASE_URL + '/objects/' + shortEmail;
+      socket.on('AddedUser', async () => {
+        const formData = new FormData();
+        setDisplayProgressBar(true);
+        formData.append('file', file);
+        let bucketURL = config.BASE_URL + '/objects/' + shortEmail;
 
-      try {
-        let response = await axios.post(bucketURL, formData, {
-          headers: {
-            authorization:
-              'Bearer ' +
-              JSON.parse(localStorage.getItem('dfs-user'))?.['token'],
-          },
-          params: {
-            inProgress: inProgressUpload.current,
-            socket_id: socket.id
-          },
-          // Added On Upload Progress Config to Axios Post Request
-          onUploadProgress: function (progressEvent) {
-            const percentCompleted = Math.round(
-              (progressEvent.loaded / progressEvent.total) * 100,
-            );
-            setProgressValue(percentCompleted);
-          },
-        });
-        if (response.status === 200) {
-          // toast.info('Upload is in Progress....Please check after some time');
-        } else {
-          toast.error('Error in Uploading File');
+        try {
+          let response = await axios.post(bucketURL, formData, {
+            headers: {
+              authorization:
+                'Bearer ' +
+                JSON.parse(localStorage.getItem('dfs-user'))?.['token'],
+            },
+            params: {
+              inProgress: inProgressUpload.current,
+              socket_id: socket.id,
+            },
+            // Added On Upload Progress Config to Axios Post Request
+            onUploadProgress: function (progressEvent) {
+              const percentCompleted = Math.round(
+                (progressEvent.loaded / progressEvent.total) * 100,
+              );
+              setProgressValue(percentCompleted);
+            },
+          });
+          if (response.status === 200) {
+            // toast.info('Upload is in Progress....Please check after some time');
+          } else {
+            toast.error('Error in Uploading File');
+          }
```

Comment:
if statement is redundant.

better

`if(response.status !== 200){...}`

Reply:
resolved